src/pydata_sphinx_theme/assets/scripts/index.js

```diff
@@ -9,6 +9,89 @@ import "bootstrap";
 
 import "../styles/index.scss";
 
+/*******************************************************************************
+ * Theme interaction
+ */
+
+var prefersDark = window.matchMedia("(prefers-color-scheme: dark)");
+
+/**
+ * set the the body theme to the one specified by the user browser
+ * @param {event} e
+ */
+function autoTheme(e) {
+  document.documentElement.dataset.theme = prefersDark.matches
+    ? "dark"
+    : "light";
+}
+
+/**
+ * Set the theme using the specified mode.
+ * It can be one of ["auto", "dark", "light"]
+ * @param {str} mode
+ */
+function setTheme(mode) {
+  if (mode !== "light" && mode !== "dark" && mode !== "auto") {
+    console.error(`Got invalid theme mode: ${mode}. Resetting to auto.`);
```

Comment:
For the future, how do you envision new light and dark modes using new themes?

Reply:
I think no more than 2 themes at the time should be used in documentation. 
If we want to add extra themes in the lib (high contrast, color-blind friendly etc...) I think we should select a different theming file at build time. For this purpose, we could split dark and light default theme in 2 files and add 2 parameters (or 1 dict) in the `conf.py` file.